### PR TITLE
[MNT] Cleanup code in aeon.util.conversion module

### PR DESCRIPTION
**Pull request by CodeLionX**
#### Reference Issues/PRs

None.

#### What does this implement/fix? Explain your changes.

When familiarizing myself with the implementation of `aeon.utils.conversion.convert_collection` for #1287, I found some smelly code and docstrings. Thus, I quickly fixed them.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Looks like we are testing for a specific error message here, will have to change that as well if the new one is kept.

File: aeon/utils/conversion/_convert_collection.py [hunk 1]
```diff
@@ -583,51 +570,51 @@ def convert_collection(X, output_type):
...
 
-def resolve_unequal_length_inner_type(inner_type):
+def resolve_unequal_length_inner_type(inner_types):
     """Hierarchy of preference for internal supported types for unequal length.
 
     Parameter
     ---------
-    inner_type: str
-        The inner type to be resolved.
+    inner_types: str
```
Comment (TonyBagnall) [inline aeon/utils/conversion/_convert_collection.py:606]: inner_types is actually a list of str

File: aeon/utils/conversion/_convert_collection.py [hunk 2]
```diff
@@ -583,51 +570,51 @@ def convert_collection(X, output_type):
...
 
-def resolve_equal_length_inner_type(inner_type):
+def resolve_equal_length_inner_type(inner_types):
     """Hierarchy of preference for internal supported types for equal length.
 
     Parameter
     ---------
-    inner_type: str
-        The inner type to be resolved.
+    inner_types: str
```
Comment (TonyBagnall) [inline aeon/utils/conversion/_convert_collection.py:578]: also a list

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
great, thanks for sorting our smelly code :) Small change on the resolve functions, they are passed lists

**Comment by TonyBagnall** [comment]
ah also, we just had a discussion about n_cases vs n_instances  n_cases there are 113 matches and n_instances 123! we think we will standardise to n_cases, its a bit shorter :)

**Review by TonyBagnall — APPROVED** [review]

**Comment by TonyBagnall** [comment]
sorry about the conflict @CodeLionX , we just put in a big refactor to make everything `n_cases` and `n_timepoints`

**Comment by CodeLionX** [comment]
> ah also, we just had a discussion about n_cases vs n_instances n_cases there are 113 matches and n_instances 123! we think we will standardise to n_cases, its a bit shorter :)

I would prefer `n_instances` because I associate the number of classes or clusters with `n_cases` instead of the number of time series within a dataset. But consistency is even more important to me 👍🏼

I reverted the renaming so that it says `n_cases` now. Please check again.

**Review by TonyBagnall — APPROVED** [review]

**Comment by CodeLionX** [comment]
I am not allowed to merge, but please go ahead.

**Pull request merged**